### ISSUE TITLE
New Task: Connect ATN and MBON into IOOS DMAC

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/marine_life_data_network#52]
### Who is requesting this?

@ioos/marine-life

### What is being requested?

Connect ATN and MBON into IOOS DMAC. Coordinate with IOOS Catalog developers (POC: @mwengren) on how ATN and/or MBON portals could be harvested for data.ioos.us. Guidance for the process to add records is documented at https://ioos.github.io/catalog/ 

### What is the current status quo (i.e., what happens if this does not get done)?

ATN and MBON datasets wont show up in data.ioos.us.
Marine Life will not meeting IOOS DMAC requirements by being discoverable in data.ioos.us.

### What indicates this is done (i.e., how do we know this is complete)?

- [ ] ATN catalog is discoverable in data.ioos.us
- [ ] MBON catalog is discoverable in data.ioos.us

### Provide a description or any other important information.

xref:
* https://github.com/ioos/ckanext-ioos-theme/issues/237
* https://github.com/ioos/catalog/issues/83

**Comment by mwengren** [comment]
Copying my comments from https://github.com/ioos/ckanext-ioos-theme/issues/237#issuecomment-2150427124 below:

AFAIK IOOS is required to furnish ISO XML metadata (or perhaps DCAT JSON, not 100% sure on that alternative) to NOAA for inclusion in NOAA's enterprise data inventories for all of our publicly-available data/services.

For all of IOOS' non-bio data, it's been fairly straightforward to do this as most of the software we use has been developed to able to output an ISO XML metadata representation of the datasets they serve. Since that isn't the case for OBIS, MBON, or ATN (I believe), that's something we'll need to address for both including those data in IOOS Catalog given its current capabilities, and also for sending up the chain to NOAA to meet requirements.

It may be that leveraging IOOS Catalog and converting the various bio data formats to ISO XML format isn't the best approach to meeting NOAA data inventory requirements.  If there are better, simpler ways to furnish these metadata to NOAA that I'm not aware, we should consider those options.   Catalog has been our solution to date, but primarily because of the pre-existing metadata format support and compatibility.

Ideally, we can have a comprehensive inventory of 100% of IOOS' data in Catalog, and I think we should still aim for that goal, but we need to understand better what the challenges for that might be wrt ATN, MBON, or other bio/Marine Life data.

**Comment by mwengren** [comment]
@MathewBiddle That makes sense on the data flow and connection in with the RA ERDDAPs, I recall that plan now... thanks for adding the example.  

I think the OBIS-USA/NCEI archive probably does meet the NOAA data publishing/open data requirements for those data - at least from what I understand.

I think our goal should be to include both access points (NCEI archive and RA ERDDAP) at the NOAA Catalog level (i.e. OneStop).  The IOOS Catalog should include all data access services provided by the RAs, or other IOOS DACs, that are funded and supported by IOOS.  

Having two separate metadata records for the same dataset should be OK as well as they'll be describing different endpoints to access the same data, presumably.   Ideally there would be a way to relate each metadata record to the other within the NOAA Catalog, but I'm not sure that is technically possible at present.  That might be a good requirement to share with the OneStop team though.  

I guess the one scenario that seems to be a potential gap where IOOS-funded bio data might not be represented in IOOS Catalog is if a provider is not serving their data via RA ERDDAP, but are aligning them to Darwin Core and submitting to NCEI.  

Ideally, we could also represent those raw data access points, whatever they might be, in IOOS Catalog as well, even if they would be technically meeting the NOAA open data publishing guidelines via OBIS/NCEI archive pathway.  

I don't know how much of a priority or how common this is... maybe would provide justification to encourage those providers to work with an RA to publish to ERDDAP, however.

**Comment by laurabrenskelle** [comment]
@mwengren Is there a reason you couldn't share the RA ERDDAP link as another data access link in the collection metadata record at NCEI? It doesn't seem ideal to have two collection records for the same dataset in OneStop. Here is an example: https://data.noaa.gov/onestop/collections/details/573b7dc1-7d06-4fdc-a134-056c112c2260

**Comment by MathewBiddle** [comment]
> I guess the one scenario that seems to be a potential gap where IOOS-funded bio data might not be represented in IOOS Catalog is if a provider is not serving their data via RA ERDDAP, but are aligning them to Darwin Core and submitting to NCEI.

I think this might be more common with cross funded efforts, like MBON. Some projects use [EDI](https://edirepository.org/) and [Arctic Data Center](https://arcticdata.io/) as their repositories (maybe BCO-DMO too).

**Comment by MathewBiddle** [comment]
This task has evolved a little bit since that last revisit in June. This activity should be initially focused on how to get ATN "data" into the IOOS Data Catalog. I put quotes around "data" because we should define what we mean by that term in the context of ATN.

For now, MBON datasets are making it to the IOOS Data Catalog via RA ERDDAP's (which are being harvested into the catalog). So, there is no effort required to make MBON datasets appear in the IOOS Data Catalog. 

Next steps:
* review https://ioos.github.io/catalog/
* Start discussing pathways for ATN data to make it to the data catalog.

**Comment by MathewBiddle** [comment]
Updating the title to be more reflective of the activity.

**Comment by MathewBiddle** [comment]
I'm understanding this task better as we continue to have conversations with @mwengren about the IOOS Data Catalog. 

The requirement for the IOOS Data Catalog is:
* A WAF: A WAF stands for a web-accessible folder. It is any folder with file contents exposed via a webserver to the outside world.
* Files contained within the WAF should be XML files conforming to the ISO 19115-2 metadata standard.

More details:
>The first step to using the Registry is to request an account, details on that can be found on the [user accounts](https://ioos.github.io/catalog/pages/registry/users) page. Once your account is approved and your IOOS organization affiliation confirmed, the next step is to configure the list of WAF and/or CS-W harvest sources that contain metadata describing your organization’s datasets. Detail on how to do this can be found on the [managing harvests](https://ioos.github.io/catalog/pages/registry/harvests) page.
>
> The Registry provides a high degree of control over managing, troubleshooting (including displaying ISO XML validation and CKAN harvesting errors), and previewing dataset metadata for publishing in the [IOOS Catalog](https://data.ioos.us/).

https://ioos.github.io/catalog/pages/registry/

**Comment by MathewBiddle** [comment]
clarified the title to what the activity is.

**Comment by MathewBiddle** [comment]
related https://github.com/ioos/marine_life_data_network/issues/44

Need to identify if we go 
1. ATN -> ERDDAP -> IOOS Data Catalog
2. ATN -> WAF -> IOOS Data Catalog

**Comment by laurabrenskelle** [comment]
Removing references of MBON from the original ticket description. The scope of this ticket is ATN metadata only.

**Comment by MathewBiddle** [comment]
The easiest solution would be:

ATN -> ERDDAP -> IOOS Data Catalog

using https://atn.ioos.us/erddap/index.html

**Comment by laurabrenskelle** [comment]
Is this issue still relevant? Do we still need a WAF for ATN metadata? If so, the NCEI ISO metadata for these records is being generated somehow, can we not leverage those?

I am moving this from ToDo to the Icebox until we have a clear idea of what should be done for this.

**Comment by MathewBiddle** [comment]
Our requirements are that all IOOS data/metadata must be discoverable through NOAA OneStop. Currently IOOS does this through the IOOS Data Catalog. However, since ATN is archiving at NCEI, and NCEI's metadata records are indexed by NOAA OneStop, we are technically meeting our requirement. 

BUT, that means that ATN records do not show up in the IOOS Data Catalog. So, what I would like to see is the ATN metadata records appearing in the IOOS Data Catalog. Yes, that means there will be duplicate records at NOAA OneStop, but I think that's okay in this case as this is similar to how all the RA non-Federal stations are working right now (archived records at NCEI and records from IOOS Data Catalog appear on NOAA OneStop).

To make everyone's life a little easier, if TT could load the netCDF files into the ATN ERDDAP, we can simply register that ERDDAP in the IOOS Data Catalog and be done. I just tested this with the IOOS Office ERDDAP and it was relatively seamless. We just have to configure the ERDDAP to produce the NOAA ISO metadata records (<useSisISO19115>false</useSisISO19115> in setup.xml).

**Comment by iwensu0313** [comment]
That sounds like a good plan - getting the data into ATN ERDDAP. For clarity that would be complete by this open task I believe https://github.com/ioos/marine_life_data_network/issues/44

**Comment by MathewBiddle** [comment]
#44 is part of it for sure. Although registering the ATN ERDDAP WAF with the IOOS Data Catalog can be done before you load data as it will harvest new records when they appear.

To register the ERDDAP in the IOOS Data Catalog:

1. Ensure the ERDDAP is generating the NOAA modified ISO 19115-2 records. This is done by setting `<useSisISO19115>false</useSisISO19115>` in setup.xml. See the changes in [version 2.29.0](https://erddap.github.io/changes#version-2290). You probably need to update the ERDDAP to the most recent version. 
1. Ensure the appropriate ISO metadata records are appearing in the ERDDAP WAF: https://atn.ioos.us/erddap/metadata/iso19115/xml/
2. Register the WAF in the IOOS Data Catalog by following the guidance at https://ioos.github.io/catalog/pages/registry/
3. Check that the harvest is appropriate https://data.ioos.us/harvest/ and records are being harvested.

**Comment by MathewBiddle** [comment]
To clarify, this task is only to register the ATN ERDDAP in the IOOS Data Catalog.

**Comment by MathewBiddle** [comment]
@iwensu0313 now that the ATN erddap is up and running, this can be picked up. See https://github.com/ioos/marine_life_data_network/issues/52#issuecomment-3959071404 for details on how to register the ERDDAP WAF with the IOOS Data Catalog.

**Comment by MathewBiddle** [comment]
Looks like the atn ERDDAP is configured to produce the correct ISO metadata records for the IOOS Data Catalog to harvest.

From https://atn.ioos.us/erddap/metrics

```
feature_flags{feature_flags="iso19115Active"} 1
feature_flags{feature_flags="useSisISO19115"} 0
feature_flags{feature_flags="useSisISO19139"} 0
```

**Comment by MathewBiddle** [comment]
@iwensu0313 any movement on this?

**Comment by MathewBiddle** [comment]
@iwensu0313 I went ahead and created a harvest source in the IOOS Data Catalog for the ATN ERDDAP. See https://data.ioos.us/harvest/about/atn-erddap-waf
